libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1719,3 +1699,190 @@ Type const& IRGeneratorForStatements::type(Expression const& _expression)
 	solAssert(_expression.annotation().type, "Type of expression not set.");
 	return *_expression.annotation().type;
 }
+
+bool IRGeneratorForStatements::visit(TryStatement const& _tryStatement)
+{
+	solAssert(!m_returnInfo.has_value(), "");
+
+	_tryStatement.externalCall().accept(*this);
+	solAssert(m_returnInfo.has_value(), "");
+	solAssert(m_returnInfo.value().functionCall == &_tryStatement.externalCall(), "");
+
+	auto const trySuccessCondition = m_context.trySuccessConditionVariable(_tryStatement.externalCall());
+
+	m_code << "switch iszero(" << trySuccessCondition << ")\n";
+
+	m_code << "case 0 { // success case\n";
+	TryCatchClause const& successClause = *_tryStatement.clauses().front();
+	decodeTryCallReturnParameters(successClause);
+	m_returnInfo.reset();
+	successClause.block().accept(*this);
+	m_code << "}\n";
+
+	m_code << "default { // failure case\n";
+	handleCatch(_tryStatement);
+	m_code << "}\n";
+
+	return false;
+}
+
+string IRGeneratorForStatements::decodeReturnParameters()
+{
+	solAssert(m_returnInfo.has_value() && m_returnInfo->functionCall != nullptr, "");
+
+	auto const retVars = IRVariable(*m_returnInfo->functionCall).commaSeparatedList();
+	if (retVars.empty())
+		return "";
+
+	return Whiskers(R"(
+		let <retVars> := <decodeReturnParameters>(<pos>, <returnSize>)
+	)")
+	("retVars", IRVariable(*m_returnInfo->functionCall).commaSeparatedList())
+	("decodeReturnParameters",
+		m_utils.decodeReturnParametersFunction(
+			m_context.revertStrings(),
+			m_returnInfo->returnTypes,
+			m_returnInfo->dynamicReturnSize,
+			retVars
+		)
+	)
+	("pos", m_returnInfo->returndataVariable)
+	("returnSize",
+		m_context.evmVersion().supportsReturndata()
+			? "returndatasize()"s
```

Comment:
Coding style

Reply:
I didn't find anything regarding ?: operator in the coding-style file but the closing bracket is now on its own line.

Reply:
Separators (including `?` and `:`) always go to the end of the line.